### ISSUE TITLE
iterative mode not working on a given dataset whereas normal mode does

### ISSUE THREAD

**Issue by etiennejourdier** [koszullab/hicstuff#68]
Hi,
I have an issue, only with a given dataset (SRR5092177), while mapping in iterative mode :
- in normal mode: >99% of the reads are aligned
- in iterative mode : <1% of the reads are aligned 

Surprisingly I do not have this issue using another very similar dataset from the same experiment (SRR5092176).

Below are the commands I used and the outputs (the fastq files were cut to 1M reads for this demo, but the result is the same with the complete fastq).

by the way, the readme.md file is not up to date, it says the option is `--iterative` instead of `--mapping=iterative`.

**With mapping = normal**
```
> hicstuff pipeline -t 4 -o out/ --prefix Rut -a bowtie2 -g RutC30.fasta fastq/SRR5092177_1.fastq.gz fastq/SRR5092177_2.fastq.gz --force
1000000 reads; of these:
  1000000 (100.00%) were unpaired; of these:
    16519 (1.65%) aligned 0 times
    888967 (88.90%) aligned exactly 1 time
    94514 (9.45%) aligned >1 times
98.35% overall alignment rate
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
1000000 reads; of these:
  1000000 (100.00%) were unpaired; of these:
    7000 (0.70%) aligned 0 times
    896849 (89.68%) aligned exactly 1 time
    96151 (9.62%) aligned >1 times
99.30% overall alignment rate
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
INFO :: Genome digested into 6978 fragments with a median length of 5000.0
INFO :: 90% reads (single ends) mapped with Q >= 30 (1793467/2000000)
INFO :: 823914 pairs used to build a contact map of 6978 bins with 240078 nonzero entries.
INFO :: Contact map generated after 0h 8m 27s
```

**With mapping = iterative**
````
> hicstuff pipeline -t 4 -o out/ --prefix Rut -a bowtie2 --mapping iterative -g RutC30.fasta fastq/SRR5092177_1.fastq.gz fastq/SRR5092177_2.fastq.gz --force
INFO :: 1000000 reads to parse
INFO :: Truncating unaligned reads to 20bp and mapping.
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/570868db/temp_alignment.bam'
INFO :: 999682 reads left to map.
INFO :: Truncating unaligned reads to 40bp and mapping again.
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/570868db/temp_alignment.bam'
INFO :: 999564 reads left to map.
INFO :: Truncating unaligned reads to 60bp and mapping again.
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/570868db/temp_alignment.bam'
INFO :: 999078 reads left to map.
INFO :: Truncating unaligned reads to 80bp and mapping again.
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/570868db/temp_alignment.bam'
INFO :: 998447 reads left to map.
INFO :: Truncating unaligned reads to 100bp and mapping again.
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/570868db/temp_alignment.bam'
INFO :: 992591 reads left to map.
INFO :: Trying to map unaligned reads at full length (101bp).
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/570868db/temp_alignment.bam'
INFO :: 992590 reads left to map.
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/570868db/temp_alignment.bam'
INFO :: 7410 reads aligned / 1000000 total reads.
INFO :: 1000000 reads to parse
INFO :: Truncating unaligned reads to 20bp and mapping.
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/1d82ba96/temp_alignment.bam'
INFO :: 999842 reads left to map.
INFO :: Truncating unaligned reads to 40bp and mapping again.
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/1d82ba96/temp_alignment.bam'
INFO :: 999749 reads left to map.
INFO :: Truncating unaligned reads to 60bp and mapping again.
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/1d82ba96/temp_alignment.bam'
INFO :: 999287 reads left to map.
INFO :: Truncating unaligned reads to 80bp and mapping again.
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/1d82ba96/temp_alignment.bam'
INFO :: 998664 reads left to map.
INFO :: Truncating unaligned reads to 100bp and mapping again.
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/1d82ba96/temp_alignment.bam'
INFO :: 992066 reads left to map.
INFO :: Trying to map unaligned reads at full length (101bp).
[bam_sort_core] merging from 0 files and 4 in-memory blocks...
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/1d82ba96/temp_alignment.bam'
INFO :: 992063 reads left to map.
[E::idx_find_and_load] Could not retrieve index file for 'out/tmp/1d82ba96/temp_alignment.bam'
INFO :: 7937 reads aligned / 1000000 total reads.
INFO :: Genome digested into 6978 fragments with a median length of 5000.0
INFO :: 1% reads (single ends) mapped with Q >= 30 (15347/2000000)
INFO :: 67 pairs used to build a contact map of 6978 bins with 64 nonzero entries.
INFO :: Contact map generated after 0h 21m 53s
````

Thanks
E.

**Comment by ABignaud** [comment]
Hi,
Thanks for the remark on the `README` I have corrected it in 5c84681ca892c1f8bc427f42ae027fd6dd7b10fb.

For the issue I manage to repeat it. It's because in the `iterative` mode it currently used the `--very-sensitive` mode (faster) and not the `--very-sensitive-local` mode as in the normal alignment mode. These mode gives lower mapping score of your mapping (i don't know why especially on this library or this genome). You can try by decreasing your mapping quality `-q 5` for example and you will recover partially your reads (only 72% mapped reads but I don't took a reference as good as you).

I have change this behavior to do local alignment in both in f0983469e56fa9c285faa1dbafb222035ba77027 to avoid this issues in the future and it resolves the issue.

**Comment by etiennejourdier** [comment]
Hi,
Thanks for the answer ! I applied the correction you made in commit f098346 and it indeed corrrected the issue.
I add the output below to show the difference -> 94% reads aligned with Q>=30
And I close the issue.
E.
```
> hicstuff pipeline -t 4 -o out/ --prefix Rut -a bowtie2 --mapping iterative -g RutC30.fasta fastq/SRR5092177_1.fastq.gz fastq/SRR5092177_2.fastq.gz --force
INFO :: 1000000 reads to parse
INFO :: Truncating unaligned reads to 20bp and mapping.
INFO :: 1000000 reads left to map.
INFO :: Truncating unaligned reads to 40bp and mapping again.
INFO :: 990222 reads left to map.
INFO :: Truncating unaligned reads to 60bp and mapping again.
INFO :: 99140 reads left to map.
INFO :: Truncating unaligned reads to 80bp and mapping again.
INFO :: 77241 reads left to map.
INFO :: Truncating unaligned reads to 100bp and mapping again.
INFO :: 64893 reads left to map.
INFO :: Trying to map unaligned reads at full length (101bp).
INFO :: 64852 reads left to map.
INFO :: 935148 reads aligned / 1000000 total reads.
INFO :: 1000000 reads to parse
INFO :: Truncating unaligned reads to 20bp and mapping.
INFO :: 1000000 reads left to map.
INFO :: Truncating unaligned reads to 40bp and mapping again.
INFO :: 989331 reads left to map.
INFO :: Truncating unaligned reads to 60bp and mapping again.
INFO :: 79543 reads left to map.
INFO :: Truncating unaligned reads to 80bp and mapping again.
INFO :: 63245 reads left to map.
INFO :: Truncating unaligned reads to 100bp and mapping again.
INFO :: 53112 reads left to map.
INFO :: Trying to map unaligned reads at full length (101bp).
INFO :: 53085 reads left to map.
INFO :: 946915 reads aligned / 1000000 total reads.
INFO :: Genome digested into 6978 fragments with a median length of 5000.0
INFO :: 94% reads (single ends) mapped with Q >= 30 (1882063/2000000)
INFO :: 903021 pairs used to build a contact map of 6978 bins with 284017 nonzero entries.
INFO :: Contact map generated after 0h 5m 31s
```
